### PR TITLE
Shorten ResNet tests and add more coverage.

### PR DESCRIPTION
**Pull request by vanbasten23**
ResNet has been a standard way to test if torch_xla works end-to-end at minimum. But it only runs in xl-ml tests which runs after PRs have been submitted and it's easy to overlook checking the xl-ml dashboard hence miss any regression. Running ResNet today in presubmit CI is also long (takes > 10min for 1 epoch with smaller fake samples). So this PR tries to shorten the ResNet test and add more in the presubmit CI. In particular:
- shortens the Resnet (eg test/test_train_mp_imagenet.py) by adding a flag just for testing, to limit the number of steps to run. To validate the functionality, we don't need to run a full epoch. Currently, the existing test `JRT_DEVICE=CUDA python test/test_train_mp_imagenet_fsdp.py --fake_data --use_nested_fsdp --use_small_fake_sample --num_epochs=1` takes 10 min to run in the CI. With the change, it takes 1 min.
- add more resnet test coverage for the critical scenarios such as MP and SPMD (single-process or mutli-process) in the presubmit CI.

### CODE REVIEW

**Comment by will-cromar** [comment]
We already run a ResNet in `test_operations.py`: https://github.com/pytorch/xla/blob/4327d24a3a9443b120ba04079a92cb5773e738cc/test/test_operations.py#L275-L307

Although I'm hesitant to start adding models to the CI, I see the value here. Can you remove the duplicate test from `test_operations.py` and use resnet18 for your functional test?

**Comment by will-cromar** [comment]
We should avoid having a resnet test defined in two places. Especially since the `test_operations` version uses the obsolete `dp.DataParallel`. IMO we should just delete the test in `test_operations` since you're adding a more relevant ResNet test to the CI.

File: .circleci/common.sh [hunk 1]
```diff
@@ -132,8 +132,17 @@ function run_torch_xla_python_tests() {
...
+        # single-host-single-process
+        PJRT_DEVICE=CUDA python3 test/test_train_mp_imagenet.py --fake_data --batch_size=16 --num_epochs=1 --num_cores=1 --num_steps=25 --model=resnet18
+
+        # single-host-multi-process
+        num_devices=$(nvidia-smi --list-gpus | wc -l)
+        PJRT_DEVICE=CUDA GPU_NUM_DEVICES=$GPU_NUM_DEVICES python3 test/test_train_mp_imagenet.py --fake_data --batch_size=16 --num_epochs=1 --num_steps=25 --model=resnet18
+        PJRT_DEVICE=CUDA torchrun --nnodes=1 --node_rank=0 --nproc_per_node=$num_devices test/test_train_mp_imagenet.py --fake_data --pjrt_distributed --batch_size=16 --num_epochs=1  --num_steps=25 --model=resnet18
+
+        # single-host-SPMD
+        XLA_USE_SPMD=1 PJRT_DEVICE=CUDA torchrun --nnodes=1 --node_rank=0 --nproc_per_node=1 test/spmd/test_train_spmd_imagenet.py --fake_data --batch_size 16 --model=resnet50 --sharding=batch --num_epochs=1  --num_steps=25 --model=resnet18
```
[the remark this answers is not in the document]
Comment (vanbasten23) [reply .circleci/common.sh:144]: It's for functionality.

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**